src/transport/AdminPairingTable.cpp

```diff
@@ -98,6 +176,64 @@ CHIP_ERROR AdminPairingInfo::GenerateKey(AdminId id, char * key, size_t len)
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR AdminPairingInfo::SetOperationalKey(const Crypto::P256Keypair & key)
+{
+    Crypto::P256SerializedKeypair serialized;
+    ReturnErrorOnFailure(key.Serialize(serialized));
+    if (mOperationalKey == nullptr)
+    {
+        mOperationalKey = chip::Platform::New<Crypto::P256Keypair>();
+    }
+    VerifyOrReturnError(mOperationalKey != nullptr, CHIP_ERROR_NO_MEMORY);
+    return mOperationalKey->Deserialize(serialized);
+}
+
+CHIP_ERROR AdminPairingInfo::SetOperationalCert(const ByteSpan & cert)
+{
+    VerifyOrReturnError(cert.size() <= kMaxChipCertSize, CHIP_ERROR_INVALID_ARGUMENT);
+    if (mRootCertLen != 0 && mRootCertLen < cert.size())
+    {
+        VerifyOrReturnError(mRootCert != nullptr, CHIP_ERROR_INCORRECT_STATE);
+        chip::Platform::MemoryFree(mRootCert);
+        mRootCertLen = 0;
+        mRootCert    = nullptr;
+    }
+
+    if (mRootCert == nullptr)
+    {
+        mRootCert = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(cert.size()));
+    }
+    VerifyOrReturnError(mRootCert != nullptr, CHIP_ERROR_NO_MEMORY);
+    VerifyOrReturnError(CanCastTo<uint16_t>(cert.size()), CHIP_ERROR_INVALID_ARGUMENT);
+    mRootCertLen = static_cast<uint16_t>(cert.size());
```

Comment:
should we have an AllocatedLen variable?

mRootCertLen is forcibly set to cert.size even if it used to be larger before